addons/actions/src/preview/withActions.ts

```diff
@@ -47,8 +50,25 @@ export const createDecorator = (actionsFn: any) => (...args: any[]) => (storyFn:
     }
     return undefined;
   }, [root, actionsFn, args]);
+}
 
+export const createDecorator = (actionsFn: any) => (...args: any[]) => (storyFn: () => any) => {
+  applyEventHandlers(actionsFn, ...args)
+  
   return storyFn();
 };
```

Comment:
createDecorator is used by the `decorate` options. I'm a bit fuzzy on that use case but we have examples of it in a few examples.